shared/process-tc-artifacts.sh

```diff
@@ -27,6 +27,17 @@ unzip -q $PLATFORM.mozsearch-index.zip -d analysis-$PLATFORM
 # rust-indexer.rs tool will take care of combining all the analysis files correctly.
 unzip -q $PLATFORM.mozsearch-rust.zip -d objdir
 
+# These zips have a rustlib top-level folder and then contain the
+# rust stdlib src (all the zips have the same src tree) and analysis
+# data for the platform. We unpack all the zips into the same
+# destination folder, implicitly merging them. And we copy the stdlib
+# src tree into the objdir and (ab)use the generated files machinery.
+unzip -qn $PLATFORM.mozsearch-rust-stdlib.zip -d .
+if [ ! -d "objdir/__RUST__" ]; then
+    mkdir -p "objdir/__RUST__"
+    cp -Rn rustlib/src/rust/src objdir/__RUST__/
```

Comment:
Do we want to do this conditionally? I assume this is not a problem when m-c updates their rust version?

Reply:
I don't expect it to be a problem when m-c updates rust, because it's still the `repack_rust.py` script that builds the rust toolchain artifacts, and that's where we patched in the code to include the stdlib src and analysis data. So it should Just Work (TM).